src/FlowDiagnostics.jl

```diff
@@ -441,6 +446,137 @@ function QVelocityGradientTensorInvariant(model; location = (Center, Center, Cen
 end
 
 const Q = QVelocityGradientTensorInvariant
-#---
+
+struct MixedLayerDepth{C}
+    criterion::C
+end
+
+"""
+    $(SIGNATURES)
+
+Returns the mixed layer depth defined as the depth at which `criterion` is true.
+
+Defaults to `DensityAnomalyCriterion` where the depth is that at which the density
+is some threshold (defaults to 0.125kg/m³) higher than the surface density.
+
+When `DensityAnomalyCriterion` is used, the arguments `buoyancy` and `C` should be 
+supplied where `buoyancy` should be the buoyancy model, and `C` should be a named 
+tuple of `(; T, S)`, `(; T)` or `(; S)` (the latter two if the buoyancy model 
+specifies a constant salinity or temperature).
+"""
+function MixedLayerDepth(grid, args...; criterion = BuoyancyAnomalyCriterion(convert(eltype(grid), -1e-4 * g_Earth)))
+    validate_criterion_model(criterion, args...)
+
+    MLD = MixedLayerDepth(criterion)
+
+    return KernelFunctionOperation{Center, Center, Nothing}(MLD, grid, args...)
+end
```

Comment:
I think the intention here is for the method to be called when `MLD = MixedLayerDepth(grid)`, right?

This isn't happening right now:
```julia
julia> @which Oceanostics.MixedLayerDepth(grid)
MixedLayerDepth(criterion::C) where C
     @ Oceanostics.FlowDiagnostics ~/repos/Oceanostics.jl2/src/FlowDiagnostics.jl:451
```
 Can confirm? If I'm right, the signature probably should be:

```suggestion
function MixedLayerDepth(grid::AbstractGrid, args...; criterion = BuoyancyAnomalyCriterion(convert(eltype(grid), -1e-4 * g_Earth)))
    validate_criterion_model(criterion, args...)

    MLD = MixedLayerDepth(criterion)

    return KernelFunctionOperation{Center, Center, Nothing}(MLD, grid, args...)
end
```

Reply:
This is also incorrect because it doesn't return `MixedLayerDepth`. I suggest defining an alias called `MixedLayerDepthOperation` and then using that as the name for the constructor. You could also change `MixedLayerDepth` to something else (like `MixedLayerDepthKernel`) and reserv ethe name `MixedLayerDepth` for the operation.